src/test/resources/registerFieldsBasicWithDocId.json

```diff
@@ -0,0 +1,134 @@
+{
+  "indexName": "test_index",
+  "field": [
+    {
+      "name": "doc_id",
+      "type": "ATOM",
+      "storeDocValues": true,
+      "docId": true
```

Comment:
Since the updateDocument method uses a Term, I think it is a requirement for the field to be indexed. Because of the wonky state of ATOM, it will always be indexed, but that is not true for all field types.
I am also a bit unsure how this works with non keyword fields. It would be good to have tests for other supported field types.

Reply:
I added tests for which field types would be supported and make sure exceptions are thrown for those which don't.